### PR TITLE
EVG-14716 Added AuthorDisplayName to Patch type

### PR DESCRIPTION
**Pull request by ziyielenachen**
[EVG-14716](https://jira.mongodb.org/browse/EVG-14716)

### Description 
Added a new field called authorDisplayName to type Patch. This will be utilized for the PatchCard on spruce so that the project patches page displays the authors' name.

### Testing 
Added a unit test

### CODE REVIEW

File: graphql/resolvers.go [hunk 1]
```diff
@@ -820,6 +820,20 @@ func (r *patchResolver) ProjectIdentifier(ctx context.Context, apiPatch *restMod
...
+	usr, err := user.FindOneById(*obj.Author)
+	if err != nil {
+		return "", ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting user from user ID: %s", err.Error()))
+	}
+	if usr == nil {
+		return "", ResourceNotFound.Send(ctx, fmt.Sprintf("Could not find user from user ID"))
+	} 
+	fmt.Printf("++++++++++++++++++++++++++++++")
+	fmt.Printf(usr.DisplayName())
+	fmt.Printf("++++++++++++++++++++++++++++++")
```
Comment (khelif96) [inline graphql/resolvers.go:833]: left over logging

File: graphql/tests/patch/queries/authorDisplayName.graphql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  patch(id: "5e4ff3abe3c3317e352062e4") {
+    id
+    author
```
Comment (khelif96) [inline graphql/tests/patch/queries/authorDisplayName.graphql:4]: These tests don't actually fetch the authorDisplayName field

**Review by khelif96 — CHANGES_REQUESTED** [review]
Great Start! Just missing something from your tests

File: graphql/resolvers.go (hunk 1 above)
Comment (ziyielenachen) [reply graphql/resolvers.go]: oops sorry haha

File: graphql/resolvers.go [hunk 3]
```diff
@@ -820,6 +820,17 @@ func (r *patchResolver) ProjectIdentifier(ctx context.Context, apiPatch *restMod
 	return utility.ToStringPtr(identifier), nil
 }
 
+func (r *patchResolver) AuthorDisplayName(ctx context.Context, obj *restModel.APIPatch) (string, error) {
+	usr, err := user.FindOneById(*obj.Author)
+	if err != nil {
+		return "", ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting user from user ID: %s", err.Error()))
+	}
+	if usr == nil {
+		return "", ResourceNotFound.Send(ctx, fmt.Sprintf("Could not find user from user ID"))
```
Comment (ablack12) [inline graphql/resolvers.go:829]: nit: you don't need fmt.Sprintf here because you aren't including any formatting

**Review by ablack12 — APPROVED** [review]

**Review by malikchaya2 — APPROVED** [review]

**Review by khelif96 — APPROVED** [review]

**Pull request merged**